### PR TITLE
Add SCCACHE_C_CUSTOM_CACHE_BUSTER to CACHED_ENV_VARS

### PR DESCRIPTION
**Pull request by t184256**
I needed a way to make hashes CWD-dependent, and it feels ugly to use other variables names for that purpose.

### CODE REVIEW

File: src/compiler/c.rs [hunk 1]
```diff
@@ -643,6 +643,7 @@ pub const CACHE_VERSION: &[u8] = b"10";
 lazy_static! {
     /// Environment variables that are factored into the cache key.
     static ref CACHED_ENV_VARS: HashSet<&'static OsStr> = [
+        "SCCACHE_C_CUSTOM_CACHE_BUSTER",
```
Comment (sylvestre) [inline src/compiler/c.rs:646]: Could you please add a comment in the code explaining the why

**Comment by t184256** [comment]
Added a comment as requested + rebased.

**Comment by sylvestre** [comment]
thinking about this a bit more, it should be also documented in https://github.com/mozilla/sccache/blob/master/README.md 
sorry

**Comment by t184256** [comment]
No problem, I tried to add a section for it and the two already existing variables.

**Pull request merged**